### DISCUSSION TITLE
[Question] Is MAUI a Microsoft thing or a Xamarin thing?

### DISCUSSION THREAD

**Discussion by WashingtonARamos** [dotnet/maui#243] [General]
I've been following the discussions for some time and two issues got me thinking for the same reason, they being:
[Is Microsoft gonna use MAUI?](https://github.com/dotnet/maui/issues/195)
and
[FluentUI Support](https://github.com/dotnet/maui/discussions/188)

The answer to FluentUI's controls implementations not being present in Xamarin (Forms or Native) is that it isn't Microsoft who supports Xamarin, rather Xamarin Corps itself (which was bought by Microsoft), which then brings me to the other issue: if Xamarin isn't exactly being supported by Microsoft's own engineers (rather being supported by the Xamarin engineers), is MAUI, the evolution of Xamarin, going to be a Microsoft thing or a Xamarin Team thing? Should we expect to see Microsoft develop applications and libraries for MAUI or is it going to remain being supported solely by Xamarin Team?
The current state of things surely tell us that Microsoft isn't bothering in creating their applications using Xamarin and, should Microsoft themselves not want to give the example showing us that MAUI is a good thing, why should we developers invest our time into it rather than follow Microsoft's example and use React (or Flutter for what matters)?

**Comment by davidortinau** [comment]
> Should we expect to see Microsoft develop applications and libraries for MAUI or is it going to remain being supported solely by Xamarin Team?

.NET MAUI is a .NET team thing, and the reason this is in `dotnet/maui`.

> why should we developers invest our time into it rather than follow Microsoft's example and use React (or Flutter for what matters)?

You tell us. We are building products for you. As we build a great product that meets your needs, with developer experiences that delight you, then we expect to see more developers everywhere using .NET MAUI.

**Comment by MappingSteve** [comment]
@WashingtonARamos 
> why should we developers invest our time into it ...

Unless you are already a passionate .NET user, I think it is much too early for anyone to attempt to persuade you.

First those of us who would LOVE to see .NET evolve into a COMPLETE cross-platform solution, EVERYWHERE, need to spend time with this as it evolves.

Given the range of platforms and technologies involved in current programming, the needed "stack", and knowledge, has become much too piece-meal. I would love to see Microsoft pull all this together, into a more coherent whole.

---------------------------------------

Regarding Xamarin, I view that as having two levels:

- underlying technologies that represent one "best practice" approach to targeting mobile. These become part of .Net 6. (This has personally been a huge advantage in targeting iOS and Android.)

- Xamarin Forms, which is a valiant effort, but ultimately falls short of its ambitious goals. I estimate that adopting it cost my company at least as much time as it saved us (2+ person-years redirected from redundant UI interaction coding to X-F related learning/customization/bugs, spread over three to five developers over the past three years) - and I'm not happy with the experience of debugging the custom controls we had to write, to get full control over look and feel, identical on iOS and Android (and next year, on UWP tablets).

What I'm hearing re the evolution to Maui sounds spot on. Specifically, being agnostic re use of MVVM vs alternatives. While I haven't had time to experiment with it yet, I am very ready to drop MVVM in favor of some flavor of MVU. Incorporating ViewModels was a net loss for us: added complexity, causing as many problems as it solved. I will be one of those pushing the envelope of MVU. As soon as I can afford the time to do so.
